test/dynamo/test_generator.py

```diff
@@ -629,13 +629,46 @@ def fn(t):
         self.assertEqual(y, t + sum(range(6)))
 
 
+class TestGeneratorSend(GeneratorTestsBase):
+    def test_send(self):
+        def double():
+            x = yield
+            yield x * 2
+
+        @torch.compile(backend="eager", fullgraph=True)
+        def fn(t):
+            gen = double()
+            next(gen)
+            return gen.send(t)
+
+        t = torch.randn(2)
+        y = fn(t)
+        self.assertEqual(y, t * 2)
+
+    def test_send_stop_iteration(self):
+        def double():
+            x = yield
+            yield x * 2
+
+        @torch.compile(backend="eager", fullgraph=True)
+        def fn(t):
+            gen = double()
+            next(gen)
+            a = gen.send(t)
+            b = gen.send(t)  # should result in StopIteration
+            return a + b
+
+        t = torch.randn(2)
+        with self.assertRaisesRegex(Unsupported, "Observed exception"):
+            fn(t)
+
```

Comment:
Can you do another test with fullgraph=False? With fullgraph=false I think this should actually raise a StopIteration

Reply:
It raises StopIteration, as you predicted.